src/driver/mod.rs

```diff
@@ -142,3 +142,17 @@ impl Entry {
         self.result
     }
 }
+
+/// Trait that allows to process all completed operations and clear all
+/// completed entries
+pub trait BatchCompleter {
```

Comment:
What is this for? Could you provide a possible usage?

Reply:
I want to process all available completions after polling and clear the completion queue because it's more efficient.

I suggest to remove `entries` parameter from `poll` signature. If we want to support `poll_one` we could require drivers to support internal to drivers `pop` method to get the oldest entries from the completion queue.

Reply:
I don't agree that this trait is more efficient. It is still an iterator and get next entry one by one. The `poll` method is at least the same as an iterator, and has potential to be faster.

Reply:
It's more about convenience and not having to deal with partially complete slices then about efficiency.